server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java

```diff
@@ -111,6 +116,8 @@ public Logger getLogger() {
 
     public static final String HARD_DELETE_BY_PROVIDER = "delete from users where identity_zone_id = ? and origin = ?";
 
+    public static final String USER_COUNT_BY_ZONE = "select count(*) from users where identity_zone_id = ? limit ?";
```

Comment:
I think that using `limit` here is wrong. It is my understanding that `select count` query always returns 1 row. `limit` limits the number of returned rows and do nothing for this query. Also looking at the code that uses this query, which requires the 2nd parameter, makes it all confusing. Remove `limit` from this query.

Reply:
for postgresql I thought it could help ,e.g. https://dba.stackexchange.com/questions/117284/count-gives-more-than-1-with-limit-1 , but yes I will remove it

Reply:
changed to int , removed limit